### PR TITLE
Vulkan: fix validation errors on Android

### LINKED ISSUE #185: Vulkan: Validation errors after enabling LunarG and Android validation layers

### PR DESCRIPTION
**Pull request by jaebaek**
Fixes #185

### CODE REVIEW

File: src/vulkan/engine_vulkan.cc [hunk 1]
```diff
@@ -51,6 +51,40 @@ VkShaderStageFlagBits ToVkShaderStage(ShaderType type) {
...
+      flags = VK_FORMAT_FEATURE_VERTEX_BUFFER_BIT;
+      break;
+    default:
+      return false;
+  }
+
+  return (properties.bufferFeatures & flags) == flags;
+}
+
+bool ExceedMaximumDescriptorSets(VkPhysicalDevice physical_device,
```
Comment (dj2) [inline src/vulkan/engine_vulkan.cc]: Let's invert this to:

IsDescriptorSetInBounds

Comment (jaebaek) [reply src/vulkan/engine_vulkan.cc]: done

File: src/vulkan/engine_vulkan.cc [hunk 2]
```diff
@@ -51,6 +51,40 @@ VkShaderStageFlagBits ToVkShaderStage(ShaderType type) {
   return VK_SHADER_STAGE_FRAGMENT_BIT;
 }
 
+bool IsFormatSupportedByPhysicalDevice(BufferType type,
+                                       VkPhysicalDevice physical_device,
+                                       VkFormat format) {
+  VkFormatProperties properties = {};
+  vkGetPhysicalDeviceFormatProperties(physical_device, format, &properties);
+
+  VkFormatFeatureFlagBits flags = VK_FORMAT_FEATURE_VERTEX_BUFFER_BIT;
```
Comment (dj2) [inline src/vulkan/engine_vulkan.cc]: This should be `flag` as there is only ever one value set.

Comment (jaebaek) [reply src/vulkan/engine_vulkan.cc]: done

**Pull request merged**